rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+
+    return ax1, ax2
+
+
+if __name__ == "__main__":
+    """Run test of code"""
+    print('Run a test of the code')
+    run_test()
+
+# DEPRECIATED FUNCTIONS HERE TO BE UPGRADED LATER
+# REPLACE update with lmfit derived function
+# from lmfit.lineshapes import gaussian
+# def fit_resolution(spectrum, xmin=-np.inf, xmax=np.inf):
+#    """Fit a Gaussian model to ['spectrum']
+#    in order to determine resolution_values.
+#
+#    Parameters
+#    ----------
+#    spectrum: array
+#        binned spectrum two column defining
+#        pixel, intensity
+#    xmin/xmax : float
+#        minimum/maximum value for fitting range
+#
+#    Returns
+#    ----------
+#    resolution : array
+#        values parameterizing gaussian function
+#        ['FWHM', 'center', 'amplitude', 'offset']
+#    """
+#    allx = spectrum[:, 0]
+#    choose = np.logical_and(allx > xmin, allx <= xmax)
+#    x = allx[choose]
+#    y = spectrum[:, 1][choose]
+#
+#    GaussianModel = lmfit.Model(gaussian)
+#    params = GaussianModel.make_params()
+#    params['center'].value = x[np.argmax(y)]
+#    params['FWHM'].set(min=0)
+#    result = GaussianModel.fit(y, x=x, params=params)
+#
+#    if result.success:
+#        resolution = [result.best_values[arg]
+#                      for arg in ['FWHM', 'center', 'amplitude', 'offset']]
+#        return resolution
+
+
+# TO BE INLCUDED LATER
+# def plot_resolution(ax2, spectrum):
+#     """ Plot blue points defining the spectrum on ax2
+#
+#     Parameters
+#     ------------
+#     ax2 : matplotlib axes object
+#     spectrum: array
+#         binned spectrum two column defining
+#         pixel, intensity
+#
+#     Returns
+#     -----------
+#     spectrum_artist : matplotlib artist
+#         Resolution plotting object
+#     """
+#     plt.sca(ax2)
+#     spectrum_artist = plt.plot(spectrum[:, 0], spectrum[:, 1], 'b.')
+#     plt.xlabel('pixels')
+#     plt.ylabel('Photons')
+#     return spectrum_artist
+#
+#
+# def plot_resolution_fit(ax2, spectrum, resolution, xmin=None, xmax=None):
+#     """Plot the gaussian fit to the resolution function
+#
+#     Parameters
+#     -----------
+#     ax2 : matplotlib axes object
+#         axes to plot on
+#     spectrum: array
+#         binned spectrum two column defining
+#         pixel, intensity
+#     xmin/xmax : float/float
+#         range of x values to plot over (the same as fitting range)
+#     resolution_values : array
+#         parameters defining gaussian from fit_resolution
+#     """
+#     plt.sca(ax2)
+#     if xmin is None:
+#         xmin = np.nanmin(spectrum[:, 0])
+#     if xmax is None:
+#         xmax = np.nanmax(spectrum[:, 0])
+#     x = np.linspace(xmin, xmax, 10000)
+#     y = gaussian(x, *resolution)
+#     return plt.plot(x, y, 'r-')
+#
+#
+#
+# DECIDE WHETHER THIS GOES HERE
+# def clean_image_threshold(photon_events, thHigh):
+#     """ Remove cosmic rays and glitches using a fixed threshold count.
+#
```

Comment:
I think we need a longer description here as well, how is this done?

Reply:
This is removed